rke/structure_rke_cluster_services_kube_api.go

```diff
@@ -314,6 +327,39 @@ func expandRKEClusterServicesKubeAPI(p []interface{}) (rancher.KubeAPIService, e
 	}
 	in := p[0].(map[string]interface{})
 
+	if v, ok := in["admission_configuration"].(string); ok && len(v) > 0 {
+		acBytes := []byte(v)
+		scheme := runtime.NewScheme()
+		err := v1.AddToScheme(scheme)
+		if err != nil {
+			return obj, fmt.Errorf("error adding to scheme: %v", err)
+		}
+		err = scheme.SetVersionPriority(v1.SchemeGroupVersion)
+		if err != nil {
+			return obj, fmt.Errorf("error setting version priority: %v", err)
+		}
+		codecs := serializer.NewCodecFactory(scheme)
+		decodedConfig := v1.AdmissionConfiguration{}
+		err = runtime.DecodeInto(codecs.UniversalDecoder(), acBytes, &decodedConfig)
+		if err != nil {
+			return obj, fmt.Errorf("error decoding admission configuration: %v", err)
+		}
+
+		// Andy B - related to https://github.com/kubernetes/kubernetes/issues/80609, runtime decoder is stripping the
+		// plugin metadata. Add it back in (would also be great if we can find out a way for the decoder to not strip
+		// data.
+		
+		// todo: add helper func?
+		for i, plugin := range decodedConfig.Plugins {
+			if plugin.Name == PodSecurityPluginName {
+				decodedConfig.Plugins[i].Configuration.APIVersion = "pod-security.admission.config.k8s.io/v1"
+				decodedConfig.Plugins[i].Configuration.Kind = "PodSecurityConfiguration"
+			}
```

Comment:
There are many kinds of plugins. Ideally, we can handle all of them, but it is out of the scope of this PR, and not an easy task to do. 

Because RKE supports two of them at the top level: PodSecurity and eventRateLimit, can you add a similar logic for handing the case of eventRateLimit? 

Example  AdmissionConfiguration for eventRateLimit:
```
apiVersion: apiserver.k8s.io/v1alpha1
kind: AdmissionConfiguration
plugins:
- configuration:
    apiVersion: eventratelimit.admission.k8s.io/v1alpha1
    kind: Configuration
    limits:
    - burst: 20000
      qps: 5000
      type: Server
  name: EventRateLimit
  path: ""
```

Reply:
To clarify, does this mean that we only need to support `PodSecurity` and `EventRateLimit` plugins for a custom admission configuration plugin? Or any type of admission configuration plugin?

Reply:
I am thinking about it, and I feel that TF should have the same level of functionality when adding support for new fields.  However, we are facing problems with decoding the AdmissionConfiguration now.   

I wonder if it will help resolve the problem if we write the flatter and expander functions ourselves for the types `AdmissionConfiguration` and `AdmissionPluginConfiguration` even though they are defined upstream. 
Another idea I am thinking about is to define local types. they are pretty much the same as the upstream, but we can define the field or yaml/json tags properly so it can be flattened and expanded properly, and then we can convert the result into/from the upstream type. 

Also, I wonder how other provider handles this problem. Is there any example we can find in GH? 

for context, RKE supports configuring `PodSecurity` and `EventRateLimit` directly at the top level in addition to the `.Services.KubeAPI.AdmissionConfiguration` field to ease the configuration. It does not mean RKE supports only those two plugins. In fact, users can configure any plugin via `.Services.KubeAPI.AdmissionConfiguration`.

Reply:
Yeah, I'm pretty dang stuck honestly.

So if RKE supports other custom AdmissionConfiguration besides PodSecurity and EventRateLimit then TF should too? But that's out of scope for now?

I've fixed the decoder issue on the TF provider end, and have added logic for AdmissionConfiguration EventRateLimit but unfortunately the config still isn't feeding correctly into rke :/ whenever rke tries to parse the cluster.yml it fails. RKE is creating this cluster.yml internally from the config passed in. I think creating local types `AdmissionConfiguration` and `AdmissionPluginConfiguration` would make updating the tf provider easier, but not resolve the parsing issue in rke.

Rancher2 defines `AdmissionConfiguration` with static json types.

Reply:
It _just_ began working!! I pushed changes :) let's discuss more in the morning.